src/Microsoft.VisualStudio.Services.Agent/JobServer.cs

```diff
@@ -142,12 +153,46 @@ public async Task<BlobIdentifierWithBlocks> UploadLogToBlobstorageService(Stream
             // Since we read this while calculating the hash, the position needs to be reset before we send this 
             blob.Position = 0;
 
-            using(var blobClient = CreateArtifactsClient(_connection, default(CancellationToken)))
+            using (var blobClient = CreateArtifactsClient(_connection, default(CancellationToken)))
             {
                 return await blobClient.UploadBlocksForBlobAsync(blobId, blob, default(CancellationToken));
             }
         }
 
+        public async Task<(DedupIdentifier dedupId, ulong length)> UploadToBlobStore(bool verbose, string itemPath, CancellationToken cancellationToken)
+        {
+            var (dedupClient, clientTelemetry) = await DedupManifestArtifactClientFactory.Instance
+                    .CreateDedupClientAsync(verbose, (str) => Trace.Info(str), this._connection, cancellationToken);
+
+            // Create chunks and identifier
+            var chunk = await ChunkerHelper.CreateFromFileAsync(FileSystem.Instance, itemPath, cancellationToken, false);
+            var rootNode = new DedupNode(new []{ chunk});
```

Comment:
Why do we need this? Doesn't the previous method return a DedupNode? https://dev.azure.com/mseng/AzureDevOps/_git/AzureDevOps?path=%2FArtifactServices%2FShared%2FBlobStore.Common%2FChunkerHelper.cs&version=GBmaster&line=39&lineEnd=40&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
There was a reason for this when we did the work on build artifacts. It may have something to do with the number of children but I do not remember.